clients/unshaded/src/main/java/tachyon/client/block/BlockStoreContext.java

```diff
@@ -156,6 +157,31 @@ public synchronized BlockWorkerClient acquireWorkerClient(String hostname) throw
   }
 
   /**
+   * Obtains a worker client to the worker with the given address.
+   *
+   * @param address the address of the worker to get a client to
+   * @return a {@link BlockWorkerClient} connected to the worker with the given hostname
+   * @throws IOException if no Tachyon worker is available for the given hostname
+   */
+  public synchronized BlockWorkerClient acquireWorkerClient(WorkerNetAddress address)
```

Comment:
*acquireWorkerClient(WorkerNetAddress), it acquires a worker client directly based on the address, it delegates to acquireLocalWorkerClient and acquireRemoteWorkerClient
acquireLocalWorkerClient(), it looks for the local worker, and then creates the client
acquireRemoteWorkerClient(String), it looks for the address of the worker of the given host name, and then creates the remote worker
acquireRemoteWorkerClient(WorkerNetAddress) it acquires the remote worker with client.

I agree there are several calls, but I think some of them are convenient methods.